master/internal/api/logs.go

```diff
@@ -0,0 +1,139 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/determined-ai/determined/master/pkg/actor"
+	"github.com/determined-ai/determined/master/pkg/logger"
+	"github.com/determined-ai/determined/proto/pkg/logv1"
+)
+
+const logCheckWaitTime = 200 * time.Millisecond
+
+// LogsRequest describes the parameters needed to target a subset of logs.
+type LogsRequest struct {
+	Offset int
+	Limit  int
+	Follow bool
+}
+
+// OnLogEntry is a callback called on each log entry.
+type OnLogEntry func(*logger.Entry) error
```

Comment:
Either make `OnLogEntry` into `OnLogEntryFn` or remove the `Fn` from `LogFetcherFn` (I'd prefer the first)